src/tests/ftest/daos_test/daos_core_test-rebuild.yaml

```diff
@@ -40,8 +40,9 @@ daos_tests:
       daos_test: r
       test_name: rebuild tests 28
       args: -s3 -u subtests="28"
-    test_r_31:
-      daos_test: r
-      test_name: rebuild tests 31
-      args: -s3 -u subtests="31"
-      test_timeout: 1500
+# disable REBUILD32 because of DAOS-2915
+#    test_r_31:
+#      daos_test: r
+#      test_name: rebuild tests 31
+#      args: -s3 -u subtests="31"
+#      test_timeout: 1500
```

Comment:
Sorry, I did not see much failure for this test recently? could you please point me the link? I did not see in DAOS-2915.

Reply:
The patch enable the the functionality of reading data from the specified shard, if not ready, it will not retry, instead, it will return failure. So before this patch landing, rebuild_31 issue is hidden. To avoid blocking the patch, I dispatch rebuild_31 temporarily.

Reply:
Means that before this patch landed, the rebuild_31 is marked as 'passed' in CI test, but it is not real pass, because the data is not from the expected shard.

Reply:
you mean data validation is not happening without this patch?  Then could you just disable shard validation, instead of disable the whole test case.  since I really need rebuild to finish in this case. I will check the validation later.

Reply:
The data validation happened in spite of whether with this patch or not. But if without this patch, the data validation may happen on other healthy shard instead of the the expected one. So this patch just discloses some existing but hidden issue. I do not know the reason for that. I do not want change rebuild test code to disable validation, just because I am afraid that it will misguide others to think that the rebuild_31 can pass the test. Then nobody will fix related issues. So if we really need such test, we can make another patch against this one to change rebuild test with disable related valuation part.

Here is some example of rebuild_31 failure:

https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-937/1/tests/

https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-938/1/tests/

Reply:
This patch (including its former ones PR#790) has been there for more than two months, it contains many DTX bugs fixing, and some other big patches depend on it. We do need to land it sooner than later.

Reply:
> you mean data validation is not happening without this patch? Then could you just disable shard validation, instead of disable the whole test case. since I really need rebuild to finish in this case. I will check the validation later.

The data verification logic is not special for rebuild_31, instead, related logic are shared by the whole rebuild test cases. If simply disable data verification, that will affect other rebuild tests. If add special logic to filter out rebuild_31, the code will looks some ugly.

Reply:
you can comment out  

// arg->rebuild_post_cb = rebuild_io_post_cb;

in rebuild_multiple_failures() ?

Reply:
OK, then please not forget to keep eyes on DAOS-2915.